OptimoveNotificationServiceExtension/Sources/OptimoveNotificationService.swift

```diff
@@ -208,40 +207,23 @@ public class OptimoveNotificationService {
         KeyValPersistenceHelper.set(newBadge, forKey: OptimobileUserDefaultsKey.BADGE_COUNT.rawValue)
     }
 
-    fileprivate class func trackDeliveredEvent(dispatchGroup: DispatchGroup, userInfo: [AnyHashable: Any], notificationId: Int) {
+    fileprivate class func trackDeliveredEvent(dispatchGroup _: DispatchGroup, userInfo: [AnyHashable: Any], notificationId: Int) {
         if isBackgroundPush(userInfo: userInfo) {
             return
         }
 
-        initializeAnalyticsHelper()
-        guard let analyticsHelper = analyticsHelper else {
-            return
-        }
-
-        let props: [String: Any] = ["type": KS_MESSAGE_TYPE_PUSH, "id": notificationId]
-
-        dispatchGroup.enter()
-
-        analyticsHelper.trackEvent(eventType: OptimobileEvent.MESSAGE_DELIVERED.rawValue, atTime: Date(), properties: props, immediateFlush: true, onSyncComplete: { _ in
-            self.syncBarrier.signal()
-        })
-
-        _ = syncBarrier.wait(timeout: .now() + .seconds(10))
-        dispatchGroup.leave()
-    }
-
-    fileprivate class func initializeAnalyticsHelper() {
-        let apiKey = KeyValPersistenceHelper.object(forKey: OptimobileUserDefaultsKey.API_KEY.rawValue)
-        let secretKey = KeyValPersistenceHelper.object(forKey: OptimobileUserDefaultsKey.SECRET_KEY.rawValue)
-
-        guard let apiKey = apiKey as? String, let secretKey = secretKey as? String else {
-            print("Extension: authorization credentials not present")
-            return
+        do {
+            let pendingAnalytics = PendingAnalyticsImpl(storage: KeyValPersistenceHelper.self)
+            try pendingAnalytics.add(
```

Comment:
when is this flushed? is it slightly changing behaviour, e.g. can it happen that message is delivered and then user never opens app, and we never receive the event?

Reply:
These metrics are not publicly available. 
We will not get these metrics if a user never opens the app.

Reply:
Add notification for standard init

Reply:
remove delivered events for now together with PendingAnalytics